PackageExplorer/Utilities/NuGetExtensions.cs

```diff
@@ -0,0 +1,271 @@
+		{
+			var result = frameworkMapping.ProfileFrameworks.SelectMany(x => x.Value).FirstOrDefault(x =>
+				string.Compare(x.DotNetFrameworkName, dotNetframeworkName, StringComparison.InvariantCultureIgnoreCase) == 0) ??
+						 frameworkMapping.ProfileOptionalFrameworks.SelectMany(x => x.Value).FirstOrDefault(x =>
+							 string.Compare(x.DotNetFrameworkName, dotNetframeworkName, StringComparison.InvariantCultureIgnoreCase) == 0);
+
+			return result;
+		}
+		#endregion Portable Framework Mappings
+
+		#region Framework Mappings
+		/// <summary>
+		/// Finds the short name of the by identifier.
+		/// </summary>
+		/// <param name="frameworkMapping">The framework mapping.</param>
+		/// <param name="frameworkIdentifier">The framework identifier.</param>
+		/// <param name="includeEquivalentFrameworks">if set to <c>true</c> [include equivalent frameworks].</param>
+		/// <returns>IEnumerable&lt;NuGetFramework&gt;.</returns>
+		public static IEnumerable<NuGetFramework> FindByIdentifierShortName(this IFrameworkMappings frameworkMapping, string frameworkIdentifier, bool includeEquivalentFrameworks = false)
+		{
+			var framework =
+				frameworkMapping.IdentifierShortNames.FirstOrDefault(
+					x =>
+						string.Compare(x.Value, frameworkIdentifier, StringComparison.InvariantCultureIgnoreCase) == 0 ||
+						frameworkIdentifier.StartsWith(x.Value, StringComparison.InvariantCultureIgnoreCase));
+
+			if (framework.Key == null) return null;
+
+			var matchIncludeVersion = char.IsDigit(frameworkIdentifier, frameworkIdentifier.Length - 1);
+
+			var result =
+				frameworkMapping.EquivalentFrameworks.Where(
+					x =>
+						(string.Compare(x.Key.Framework, framework.Key, StringComparison.InvariantCultureIgnoreCase) == 0 &&
+						 string.Compare(x.Key.ToShortName(matchIncludeVersion), frameworkIdentifier,
+							 StringComparison.InvariantCultureIgnoreCase) == 0) ||
+						string.Compare(x.Value.Framework, framework.Key, StringComparison.InvariantCultureIgnoreCase) == 0 &&
+						string.Compare(x.Value.ToShortName(matchIncludeVersion), frameworkIdentifier,
+							StringComparison.InvariantCultureIgnoreCase) == 0)
+					.Select(x => x).ToList();
+
+			if (includeEquivalentFrameworks)
+			{
+				return result.Select(x => x.Key).Concat(result.Select(x => x.Value));
+			}
+
+			return result.Select(x => x.Value);
+		}
+
+		/// <summary>
+		/// Finds the name of the by framework.
+		/// </summary>
+		/// <param name="frameworkMapping">The framework mapping.</param>
+		/// <param name="frameworkName">Name of the framework.</param>
+		/// <returns>NuGetFramework.</returns>
+		public static NuGetFramework FindByFrameworkName(this IFrameworkMappings frameworkMapping, string frameworkName)
+		{
+			var result = frameworkMapping.EquivalentFrameworks.FirstOrDefault(x =>
+				string.Compare(x.Key.Framework, frameworkName, StringComparison.InvariantCultureIgnoreCase) == 0 ||
```

Comment:
equals is case and locale sensitive -- Compares is more of a "best practice" from a development perspective.  Since the NuGet packages have the potential to be localized, I didn't want anything to break :)